### ISSUE TITLE
Implement opportunistic locking

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#366]
Originally reported of Google Code with ID 310.

```
Locksets need to gain the ability to be acquired opportunistically to improve instance allocations.
```
Originally added on 2012-11-15 16:03:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 1f1188c
Author: Michael Hanselmann <hansmi@google.com>
Date:   Fri Dec 7 13:29:49 2012 +0100

    cmdlib: Opportunistic locking on instance creation
    
    Adds a new parameter to “OpInstanceCreate” and “OpInstanceMultiAlloc” to
    use opportunistic locks.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2012-12-07 12:52:39 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 78bfb56
Author: Michael Hanselmann <hansmi@google.com>
Date:   Tue Dec 4 15:53:55 2012 +0100

    cmdlib: Use locked nodes as node whitelist
    
    Also actually start using opportunistic locks (if requested).
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2012-12-10 11:42:57 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
